plugins/HeightProfile.jsx

```diff
@@ -300,19 +300,23 @@ class HeightProfile extends React.Component {
                 nodes.unshift({x: x[0], y: y[0]});
                 nodes.push({x: x[x.length - 1], y: y[y.length - 1]});
 
-                return {
-                    dataset: entry.dataset,
-                    x: x,
-                    y: elevations,
-                    maxY: Math.max(...elevations),
-                    totLength: totLength,
-                    nodes: nodes
-                };
-            });
+                if (!elevations.every(val => val === 0)) {
+                    return {
+                        dataset: entry.dataset || `Dataset ${index + 1}`,
```

Comment:
The fallback should be localized.

Reply:
Yes, or we could actually force via elevation service config schema that a dataset is named.